### PR TITLE
P81-20072 - improve printing errors in listConn and msg

### PR DESCRIPTION
**Pull request by lotanaharoniP81**

### CODE REVIEW

File: listConns.go [hunk 1]
```diff
@@ -9,9 +9,9 @@ func (c *ClientConn) ListConns(ike string) (conns []map[string]IKEConf, err erro
 	var eventErr error
 
 	defer func() {
-		err = c.UnregisterEvent("list-conn")
-		if err != nil {
-			err = fmt.Errorf("error unregistering list-conns event: %v", err)
+		errUnregister := c.UnregisterEvent("list-conn")
+		if errUnregister != nil {
+			err = fmt.Errorf("error unregistering list-conns event: %v. original error: %v", errUnregister, err)
```
Comment (YafimK) [inline listConns.go]: why we arent using error wrapping? with %w? 
what happens if err is empty.

File: msg.go [hunk 2]
```diff
@@ -63,7 +63,7 @@ type segment struct {
 	msg  map[string]interface{}
 }
 
-//msg 在内部以下列3种类型表示(降低复杂度)
+// msg 在内部以下列3种类型表示(降低复杂度)
```
Comment (YafimK) [inline msg.go:66]: didn't know you know Chinese.

File: msg.go [hunk 3]
```diff
@@ -93,13 +93,13 @@ func writeSegment(w io.Writer, msg segment) (err error) {
 	//写长度
 	err = binary.Write(w, binary.BigEndian, uint32(buf.Len()))
 	if err != nil {
-		fmt.Printf("[writeSegment] error writing to binary \n")
+		fmt.Printf("[writeSegment] error writing to binary: %v \n", err)
```
Comment (YafimK) [inline msg.go:96]: why aren't we returning this?

File: msg.go [hunk 4]
```diff
@@ -93,13 +93,13 @@ func writeSegment(w io.Writer, msg segment) (err error) {
...
 		return
 	}
 
 	_, err = buf.WriteTo(w)
 	if err != nil {
-		fmt.Printf("[writeSegment] error writing to buffer \n")
+		fmt.Printf("[writeSegment] error writing to buffer: %v \n", err)
```
Comment (YafimK) [inline msg.go:102]: why aren't we returning this?

**Review by ranamarP81 — APPROVED** [review]

**Review by NoamWeiler — APPROVED** [review]

File: listConns.go (hunk 1 above)
Comment (lotanaharoniP81) [reply listConns.go]: Fixed

**Pull request merged**